File: microservices/jobofferservice/src/main/java/com/xwsdislinkt/jobofferservice/Model/JobOffer.java
package com.xwsdislinkt.jobofferservice.Model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "job_offers")
public class JobOffer {
    @Id
    public String id;
    public String position;
    public String jobDescription;
    public String requirements;
    public String publisherId;
    public String companyId;
    public Boolean isPublished;

}

File: frontend/dislinkt-xws-front/src/app/model/jobOffer.ts
export interface JobOffer {
    id: string,
    position: string,
    jobDescription: string,
    requirements: string,
    companyId: string,
    isPublished: boolean
}

export interface JobOfferDTO {
    id: string,
    position: string,
    jobDescription: string,
    requirements: string,
    publisherId: string,
    companyId: string,
    isPublished: boolean
}